esphome/components/deep_sleep/deep_sleep_bk72xx.cpp

```diff
@@ -0,0 +1,70 @@
+#ifdef USE_BK72XX
+
+#include "deep_sleep_component.h"
+#include "esphome/core/application.h"
+#include "esphome/core/log.h"
+#include <cinttypes>
+
+namespace esphome {
+namespace deep_sleep {
+
+static const char *const TAG = "deep_sleep";
+
+optional<uint32_t> DeepSleepComponent::get_run_duration_() const { return this->run_duration_; }
+
+void DeepSleepComponent::dump_config_platform_() {
+  if (wakeup_pins_.size() > 0) {
+    for (WakeUpPinItem item : this->wakeup_pins_) {
+      LOG_PIN("  Wakeup Pin: ", item.wakeup_pin);
+    }
+  }
+}
+
+bool DeepSleepComponent::pin_prevents_sleep_(WakeUpPinItem &pinItem) const {
+  return (pinItem.wakeup_pin_mode == WAKEUP_PIN_MODE_KEEP_AWAKE && pinItem.wakeup_pin != nullptr &&
+          !this->sleep_duration_.has_value() && (pinItem.wakeup_level == get_real_pin_state_(*pinItem.wakeup_pin)));
+}
+
+bool DeepSleepComponent::prepare_to_sleep_() {
+  if (wakeup_pins_.size() > 0) {
+    for (WakeUpPinItem &item : this->wakeup_pins_) {
```

Comment:
```suggestion
    for (const WakeUpPinItem &item : this->wakeup_pins_) {
```

compiler probably does it already anyways but nicer to be explict

Reply:
can't be const here

Reply:
Yeah underlying api would need a breaking change.. unfortunate